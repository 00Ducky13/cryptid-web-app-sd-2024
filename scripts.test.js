//import {screen,render} from '@testing-library/dom';
//import userEvent from '@testing-library/user-event';

//document.body.innerHTML = html;
//let board = document.getElementById('board');
//let buildingsBoard = document.getElementById('buildings');
//let tokenBoard = document.getElementById('tokens');
//var buildingsBoard = document.createElement('canvas');
//const ctxBoard = board.getContext('2d');
//const ctxBuild = buildings.getContext('2d');
//const ctxTokens = tokens.getContext('2d');
import html from './index.html';
beforeEach(() => {
    // Set up the DOM here
    
    document.body.innerHTML = html;
    HTMLDialogElement.prototype.show = jest.fn();
    HTMLDialogElement.prototype.showModal = jest.fn();
    HTMLDialogElement.prototype.close = jest.fn();
    

  });

test('testJestSetup', ()=> {
    const thisIsTrue = true
    expect(thisIsTrue).toBe(true)
  })

  test('createPoly_sixSides_outputContainsSixElements', ()=> {
    const board = document.getElementById('board');
    const ctxBoard = board.getContext('2d');
    const TAU = 2 * Math.PI;
    const RADIUS = 30;
    const scripts = require('./scripts');
    
    expect(scripts.createPoly(6,RADIUS).length).toBe(6);
  })
