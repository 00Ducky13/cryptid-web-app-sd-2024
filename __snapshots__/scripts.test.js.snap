// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`drawBoard drawBoard_basicBoardNoBuildings_correctDrawOnCanvasAndSavedCorrectInElementsArr 1`] = `Array []`;

exports[`drawBoard drawBoard_basicBoardNoBuildings_editElementsCorrect 1`] = `Array []`;

exports[`drawBoard drawBoard_randomIntroBoard_correctDrawOnCanvasAndSavedCorrectInElementsArr 1`] = `Array []`;

exports[`drawBoard drawBoard_randomIntroBoard_editElementsCorrect 1`] = `Array []`;

exports[`drawBoard drawBoard_randomNormalBoard_correctDrawOnCanvasAndSavedCorrectInElementsArr 1`] = `Array []`;

exports[`drawBoard drawBoard_randomNormalBoard_editElementsCorrect 1`] = `Array []`;

exports[`drawBuild drawBuild_blackShack11_buildingDrawnOnCanvasAndSavedInElementsArr 1`] = `Array []`;

exports[`drawBuild drawBuild_blackStone11_buildingDrawnOnCanvasAndSavedInElementsArr 1`] = `Array []`;

exports[`drawBuild drawBuild_blueShack11_buildingDrawnOnCanvasAndSavedInElementsArr 1`] = `Array []`;

exports[`drawBuild drawBuild_blueStone11_buildingDrawnOnCanvasAndSavedInElementsArr 1`] = `Array []`;

exports[`drawBuild drawBuild_greenShack11_buildingDrawnOnCanvasAndSavedInElementsArr 1`] = `Array []`;

exports[`drawBuild drawBuild_greenStone11_buildingDrawnOnCanvasAndSavedInElementsArr 1`] = `Array []`;

exports[`drawBuild drawBuild_whiteShack11_buildingDrawnOnCanvasAndSavedInElementsArr 1`] = `Array []`;

exports[`drawBuild drawBuild_whiteStone11_buildingDrawnOnCanvasAndSavedInElementsArr 1`] = `Array []`;

exports[`drawBuildings drawBuildings_blackStone11blackShack44_buildingDrawnOnCanvasAndSavedInElementsArr 1`] = `Array []`;

exports[`drawBuildings drawBuildings_blueStone11blueShack44_buildingDrawnOnCanvasAndSavedInElementsArr 1`] = `Array []`;

exports[`drawBuildings drawBuildings_greenStone11greenShack44_buildingDrawnOnCanvasAndSavedInElementsArr 1`] = `Array []`;

exports[`drawBuildings drawBuildings_whiteStone11whiteShack44_buildingDrawnOnCanvasAndSavedInElementsArr 1`] = `Array []`;

exports[`drawGrid drawGrid_tileOrder123456_canvasAtStartHexesCorrectColor 1`] = `Array []`;

exports[`drawGrid drawGrid_tileOrder123456AllFlips_canvasAtStartHexesCorrectColor 1`] = `Array []`;

exports[`drawGrid drawGrid_tileOrder123456NoFlips_canvasAtStartHexesCorrectColor 1`] = `Array []`;

exports[`drawHabs drawHabs_tileOrder123456AllFlips_elementsArrCorrectHabs 1`] = `Array []`;

exports[`drawHabs drawHabs_tileOrder123456NoFlips_elementsArrCorrectHabs 1`] = `Array []`;

exports[`drawPoly drawPoly_hexCenteredOn11_canvasAt11ColoredBlackOnBoard 1`] = `
Array [
  Object {
    "props": Object {
      "value": "#ffffff",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillStyle",
  },
  Object {
    "props": Object {
      "value": 1,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  Object {
    "props": Object {
      "value": "#000000",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
]
`;

exports[`drawPoly drawPoly_hexCenteredOn11_canvasAt11ColoredBlueOnBoard 1`] = `
Array [
  Object {
    "props": Object {
      "value": "rgba(0, 195, 217, 0.75)",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillStyle",
  },
  Object {
    "props": Object {
      "value": 1,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  Object {
    "props": Object {
      "value": "#000000",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
]
`;

exports[`drawTile drawTile_tile1CenteredOn11_canvasAt11ColorEqualToWater 1`] = `Array []`;

exports[`drawTile drawTile_tile1CenteredOn11_canvasAt42ColorEqualToDesert 1`] = `Array []`;

exports[`drawTile drawTile_tile1CenteredOn14_canvasAt14ColorEqualToWater 1`] = `Array []`;

exports[`drawTile drawTile_tile1CenteredOn17_canvasAt17ColorEqualToWater 1`] = `Array []`;

exports[`drawTile drawTile_tile1CenteredOn71_canvasAt71ColorEqualToWater 1`] = `Array []`;

exports[`drawTile drawTile_tile1CenteredOn74_canvasAt74ColorEqualToWater 1`] = `Array []`;

exports[`drawTile drawTile_tile1CenteredOn77_canvasAt77ColorEqualToWater 1`] = `Array []`;

exports[`drawTile drawTile_tile2CenteredOn11_canvasAt11ColorEqualToSwamp 1`] = `Array []`;

exports[`genEmptyBoard genEmptyBoard_none_correctDrawOnCanvasAndSavedCorrectInElementsArr 1`] = `Array []`;
